[pytorch] remove unused flags from code analyzer & move format support to python

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37404 [pytorch] add '__BASE__' section to op deps to factor out frequently used util ops
* **#37393 [pytorch] remove unused flags from code analyzer & move format support to python**

Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

Differential Revision: [D21280836](https://our.internmc.facebook.com/intern/diff/D21280836)

ljk53 committed:
[pytorch] remove unused flags from code analyzer & move format support to python

Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] remove unused flags from code analyzer & move format support to python"


Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] remove unused flags from code analyzer & move format support to python"


Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] remove unused flags from code analyzer & move format support to python"


Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

Differential Revision: [D21280836](https://our.internmc.facebook.com/intern/diff/D21280836)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM

bhosmer commented on tools/code_analyzer/op_deps_processor.py:
```diff
@@ -0,0 +1,109 @@
+        return yaml.safe_load(stream)
+
+
+def convert(fname, graph, OUTPUT, OP, OP_DEP):
```
nit: might be slightly easier to distinguish between these params and the template globals if the params were lowercase

ljk53 replied:
hmm, on the other hand I want to distinguish between code template v.s. code after substitution; maybe I can use "_template" suffix instead...

bhosmer replied:
ooo yeah, `_template` is perfect

ljk53 committed:
Update on "[pytorch] remove unused flags from code analyzer & move format support to python"


Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

Differential Revision: [D21280836](https://our.internmc.facebook.com/intern/diff/D21280836)

[ghstack-poisoned]
